lib/data_magic/document_builder.rb

```diff
@@ -0,0 +1,140 @@
+require './lib/expression'
+
+module DataMagic
+  module DocumentBuilder
+    class << self
+      def logger
+        DataMagic::Config.logger
+      end
+
+      # parse a row from a csv file, returns a nested document
+      # row: a hash  { field => value } where all values are strings
+      # fields: column_name => field_name
+      # config: DataMagic.Config instance for dictionary, column types, NULL
+      def parse_row(row, fields, config, options={}, additional=nil)
```

Comment:
agreed.  Now that we have some unit tests, it is my hope this is not in the far distant future, but not this week!